src/react/components/pages/projectSettings/projectSettingsPage.tsx

```diff
@@ -85,7 +86,7 @@ export default class ProjectSettingsPage extends React.Component<IProjectSetting
     private onFormSubmit = async (project: IProject) => {
         const isNew = !(!!project.id);
 
-        await this.ensureSecurityToken(project);
+        await this.props.applicationActions.ensureSecurityToken(this.props.appSettings, project);
```

Comment:
Instead of loading in the `appSettings` into this component we can have the redux action lookup that appSettings state during this action.